### PR TITLE
feat: introduce Tool Header component from BTP design

### PR DESCRIPTION
**Pull request by InnaAtanasova**
## Related Issue
Closes SAP/fundamental-styles#2209

## Description
This PR introduces Tool Header component  from BTP design

#### Please check whether the PR fulfills the following requirements

1. The output matches the design specs
- [x] All values are in `rem`
- [x] Text elements follow the truncation rules
- [x] hover state of the element follow design spec
- [x] focus state of the element follow design spec
- [x] active state of the element follow design spec
- [x] selected state of the element follow design spec
- [x] selected hover state of the element follow design spec
- [x] pressed state of the element follow design spec
- [x] Responsiveness rules - the component has modifier classes for all breakpoints
- [x] Includes Compact/Cosy/Tablet design
- [x] RTL support
2. The code follows fundamental-styles code standards and style
- [x] only one top level `fd-*` class is used in the file
- [x] BEM naming convention is used
- [x] Mixins are used for repeatable code (`fd-rtl`, `fd-ellipsis`, `fd-flex`, `fd-selected`, `fd-focus`, ect.)
- [x] A11y support - keyboard support, screenreader support, proper ARIA attributes, etc.
- [x] `fd-reset()` mixin is applied to all elements
- [x] Variables are used, if some value is used more than twice.
- [x] Checked if current components can be reused, instead of having new code.
3. Testing
- [x] tested Storybook examples with "CSS Resources" `normalize` option 
- [x] tested Storybook examples with "CSS Resources" `unnormalize` option 
- [x] Verified all styles in IE11
- [x] Updated tests
- [x] last commit message should have `[ci visual]` so it can trigger chromatic visual regression (e.g. `test: run chromatic visual regression [ci visual]`)
4. Documentation
- [x] Storybook documentation has been created/updated
- [na] Breaking Changes [wiki](https://github.com/SAP/fundamental-styles/wiki/Breaking-Changes) has been updated in case of breaking changes.

### CODE REVIEW

File: src/tool-header.scss [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+    }
+  }
+
+  &__logo {
+    @include fd-reset();
+
+    width: auto;
+    height: 1.625rem;
+    object-fit: cover;
+    background-size: contain;
```
Comment (N1XUS) [inline src/tool-header.scss:112]: Maybe add a modifier for 'cover' option?

File: src/tool-header.scss [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+
+        &[class*="inverted"] {
+          text-shadow: none;
+          color: map_get($set-params, 'iconColorInverted');
+
+          [class*="icon"] {
+            color: map_get($set-params, 'iconColorInverted');
+          }
+
+          background: map_get($set-params, 'bgColorInverted');
```
Comment (N1XUS) [inline src/tool-header.scss:221]: Maybe move this block into mixin?

File: src/tool-header.scss [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+      padding: map_get($set-padding, 'padding');
+    }
+  }
+
+  &--responsive-paddings {
+    @media (max-width: $fd-tool-header-screen-size-sm) {
+      @include fd-set-paddings-x-equal(1rem);
+    }
+
+    @media (min-width: 600px) and (max-width: 1023px) {
```
Comment (N1XUS) [inline src/tool-header.scss:283]: Can we have these responsive breakpoints moved into some global variables for all styles, so we keep breakpoints as consistent as possible?

**Review by N1XUS — COMMENTED** [review]
LGTM

**Review by Smiranin — APPROVED** [review]

File: src/tool-header.scss (hunk 1 above)
Comment (InnaAtanasova) [reply src/tool-header.scss:112]: it's actually not needed, removed it.

File: src/tool-header.scss (hunk 2 above)
Comment (InnaAtanasova) [reply src/tool-header.scss:221]: Mixins can not be defined within control directives or other mixins. In this case we are inside an `@each` loop and rely on variables like `$set-params` which can't be accessed outside the loop.

**Comment by InnaAtanasova** [comment]
@N1XUS thanks for the review. Your comments have been addressed. Please check again.

**Pull request merged**